tests/test_context.py

```diff
@@ -48,6 +48,19 @@ def nothing(data, r, t):
     assert np.all(r == r_changed)
 
 
+def test_byte_strings_as_run_id():
+    with tempfile.TemporaryDirectory() as temp_dir:
+        st = strax.Context(storage=strax.DataDirectory(temp_dir,
+                                                       deep_scan=True),
+                           register=[Records])
+
+
+        records_bytes = st.get_array(b'0', 'records')
+        records = st.get_array('0', 'records')
+        assert np.all(records_bytes == records)

```

Comment:
👍 thanks for checking. The only place I'm a bit concerned is for making masks based on run_id since:
```python
 b'1' == '1' -> False
```
Hence I opened issue #598 as I felt I might have to think a bit more about it